sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.digitaltwins.core;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of {@link DigitalTwinsClient
+ * DigitalTwinsClients} and {@link DigitalTwinsAsyncClient DigitalTwinsAsyncClients}, call {@link #buildClient() buildClient} and {@link
+ * #buildAsyncClient() buildAsyncClient} respectively to construct an instance of the desired client.
+ */
+@ServiceClientBuilder(serviceClients = {DigitalTwinsClient.class, DigitalTwinsAsyncClient.class})
+public class DigitalTwinsClientBuilder {
+    private final List<HttpPipelinePolicy> additionalPolicies = new ArrayList<>();
+    // mandatory
+    private String endpoint;
+    private TokenCredential tokenCredential;
+    // optional/have default values
+    private DigitalTwinsServiceVersion serviceVersion;
+    private HttpPipeline httpPipeline;
+    private HttpClient httpClient;
+    private HttpLogOptions logOptions;
+    private RetryPolicy retryPolicy;
+
+    private static HttpPipeline buildPipeline(TokenCredential tokenCredential, String endpoint,
+                                              HttpLogOptions logOptions, HttpClient httpClient,
+                                              List<HttpPipelinePolicy> additionalPolicies, RetryPolicy retryPolicy) {
+        // Closest to API goes first, closest to wire goes last.
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new RequestIdPolicy());
+
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+        policies.add(retryPolicy);
+
+        policies.add(new AddDatePolicy());
+        HttpPipelinePolicy credentialPolicy = new BearerTokenAuthenticationPolicy(tokenCredential, GetAuthorizationScopes(endpoint));
+        policies.add(credentialPolicy);
+        policies.addAll(additionalPolicies);
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
```

Comment:
A comment on what is added before retry policy, and what is added after would help as well